pipeline-runner/R/gem2s-1-download_cellranger_files.R

```diff
@@ -1,29 +1,36 @@
-download_cellranger <- function(input, pipeline_config) {
+#' Download cellranger files from S3
+#'
+#' @param input The input object from the request
+#' @param pipeline_config result of \code{load_config}
+#' @param prev_out \code{list()} that is added to in each with gem2s task.
+#'
+#' @return list where 'output' slot has config used by \code{load_cellranger}
+#' @export
+#'
+download_cellranger_files <- function(input, pipeline_config, prev_out = list()) {
   project_id <- input$projectId
   sample_names <- input$sampleNames
   sample_uuids <- input$sampleIds
 
-  print_config(1, "Cellranger", input, pipeline_config, list())
-
   s3 <- paws::s3(config = pipeline_config$aws_config)
 
-  fnames <- c("features.tsv.gz", "barcodes.tsv.gz", "matrix.mtx.gz")
+  cellranger_fnames <- c("features.tsv.gz", "barcodes.tsv.gz", "matrix.mtx.gz")
   unlink("/input", recursive = TRUE)
 
   for (sample in sample_uuids) {
-    for (fname in fnames) {
-      gem_key <- file.path(project_id, sample, fname)
+    message("\nSample --> ", sample)
+
+    for (cellranger_fname in cellranger_fnames) {
+      gem_key <- file.path(project_id, sample, cellranger_fname)
 
-      message("GEM key")
-      message(gem_key)
+      message("GEM key: ", gem_key)
 
       sample_name <- sample_names[[match(sample, sample_uuids)]]
```

Comment:
Is it possible to have `match` executed on `sample` that is not in `sample_uuids`? What will happen here in the code if this happens?

Reply:
Not possible as `sample` is from interating through `sample_uuids`

Reply:
I see